lib/ReactContexts/DataAttributionContext.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {
+  Context as ReactContext,
+  createContext,
+  FC,
+  useCallback,
+  useContext,
+  useMemo
+} from "react";
+
+interface IDataAttributionContext {
+  dataAttributionVisible: boolean;
+  showDataAttribution: () => void;
+  hideDataAttribution: () => void;
+}
+
+const Context = createContext<IDataAttributionContext | null>(null);
+
+export const DataAttributionContextProvider: FC = ({ children }) => {
+  const [dataAttributionVisible, setDataAttributionVisible] = React.useState(
+    false
+  );
+
+  const showDataAttribution = useCallback(() => {
+    setDataAttributionVisible(true);
+  }, [setDataAttributionVisible]);
+
+  const hideDataAttribution = useCallback(() => {
+    setDataAttributionVisible(false);
+  }, [setDataAttributionVisible]);
+
+  const contextValue = useMemo(
+    () => ({
+      dataAttributionVisible,
+      showDataAttribution,
+      hideDataAttribution
+    }),
+    [dataAttributionVisible, showDataAttribution, hideDataAttribution]
+  );
+  return <Context.Provider value={contextValue}>{children}</Context.Provider>;
+};
+
+export const useDataAttributionContext = (): IDataAttributionContext => {
+  const context = useContext<IDataAttributionContext>(
+    Context as ReactContext<IDataAttributionContext>
+  );
+  if (!context) {
+    throw new Error(
+      "useDataAttributionContext must be used within a DataAttributionContextProvider"
+    );
+  }
+  return context;
+};
```

Comment:
I dont think we are using ReactContext anywhere else in Terriajs. Could this functionality (dataAttributionVisible, showDataAttribution, hideDataAttribution) put in one of the global state classes e.g. ViewState instead? Once again, let us know if there are benefits to adopting this pattern instead, i.e. avoiding prop drilling. I appreciate it works well but just for consistency sake we should use the same patterns throughout the repo as much as possible.

Reply:
Yeah prop drilling and using react state for what it is meant to be used as there is no reason to make this observables and IMHO, the ViewState is toooooo big (the same as `Terria.ts`)
But I can put it in viewState, if that is team call

Reply:
@steve9164 what are your thoughts on this point?